File: testsystem-ui/src/app/model/TestSession.ts
import {Question} from "./Question";

export class TestSession {

    testId: string;
    userId: string;
    userName: string;
    startTime: Date;
    durationMinutes: number;
    elapsedTime: number;
    score: number;
    questionSnapshots: Question[];
    userAnswersByQuestionId: Map<number, string[]>
}

File: testsystem-service/src/main/java/tk/tcomad/testsystem/model/domain/TestSession.java
package tk.tcomad.testsystem.model.domain;

import java.time.Instant;
import java.util.Map;
import java.util.Set;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class TestSession {

    private Long id;
    private String testId;
    private Integer durationMinutes;
    private String userId;
    private String userName;
    private Instant startTime;
    private Integer elapsedTime;
    private Float score;
    private Set<Question> questionSnapshots;
    private Map<Long, Set<String>> userAnswersByQuestionId;
}
